src/glass/src/app/pages/deployment-page/deployment-page.component.ts

```diff
@@ -37,43 +36,47 @@ export class DeploymentPageComponent implements OnInit {
     private dialog: MatDialog,
     private notificationService: NotificationService,
     private orchService: OrchService,
-    private services: ServicesService
+    private services: ServicesService,
+    private pollService: PollService
   ) {}
 
   ngOnInit(): void {
-    this.startBlockUI('Please wait, fetching device information ...');
-    this.getDevices(0);
+    this.getDevices();
     this.updateCephfsList();
   }
 
   addNfs(): void {
     this.nfs = true;
   }
 
-  getDevices(attempt: number): void {
-    const maxAttempts = 5;
-    of(true)
+  getDevices(): void {
+    this.startBlockUI('Please wait, fetching device information ...');
+    this.orchService
+      .devices()
       .pipe(
-        delay(5000),
-        mergeMap(() => this.orchService.devices())
+        this.pollService.poll(
+          (hostDevices) => {
+            let hasDevices = false;
+            Object.values(hostDevices).forEach((v) => {
```

Comment:
I think this should also work.
```suggestion
            let hasDevices = Object.values(hostDevices).some((v) => v.devices.length);
```